_source/_posts/2021-12-06-spring-data-elasticsearch.md

```diff
@@ -0,0 +1,369 @@
+
+How can be Elasticsearch integrated into a Spring Boot Application? What are the options for adding Elasticsearch to a Java application? This post will give you a quick introduction to Elasticsearch integration options.
+
+**This tutorial was created with the following frameworks and tools**:
+
+- [HTTPie 3.0.2](https://httpie.io/)
+- [JHipster 7.6.0](https://www.jhipster.tech/installation/)
+- [Java OpenJDK 11](https://jdk.java.net/java-se-ri/11)
+- [Okta CLI 0.9.0](https://cli.okta.com)
+- [Docker 20.10.7](https://docs.docker.com/engine/install/)
+
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+## Elasticsearch integration options for Spring Boot
+
+For Java applications including Spring Boot applications, Elasticsearch provides the following [clients](https://www.elastic.co/guide/en/elasticsearch/client/index.html) for integration:
+
+- [Java Transport Client](https://www.elastic.co/guide/en/elasticsearch/client/java-api/current/index.html): Deprecated in Elasticsearch 7.0.0. Provides a client object to execute all operations asynchronously, accepting a listener or returning a future.
+
+- [Java REST Client](https://www.elastic.co/guide/en/elasticsearch/client/java-rest/7.15/index.html): Composed by the Low Level REST Client and the High Level REST Client. The Low Level Client provides load balancing, failover, persistent connections, and request/response trace logging. The High Level Client works on top of the Low Level Client and is the replacement for the `TransportClient`. It depends on the Elasticsearch core and provides synchronous and asynchronous APIs.
+
+- [Java API Client](https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/index.html): The new client library, independent from Elasticsearch core, provides strongly typed requests and responses, blocking and asynchronous versions for all APIs, fluent builders and functional patterns, jackson and JSON-b support.
+
+## Hello Spring Data Elasticsearch!
+
+Spring Data Elasticsearch is another integration option that adds the Spring Repository abstraction at the search layer, providing access and search functionality for domain objects stored in Elasticsearch. Operations are sent through a client connected to the Elasticsearch node. With Spring Data, the High Level REST Client is the default client, although Elasticsearch documentation states that this client is deprecated in favor of the Java API Client since version 7.15. The Java API Client is not listed as a supported client yet. The Java Transport Client is still supported in Spring Data, but the general recommendation is to use the High Level Client.
```

Comment:
```suggestion
Spring Data Elasticsearch is another integration option that adds the Spring Repository abstraction at the search layer, providing access and search functionality for domain objects stored in Elasticsearch. Operations are sent through a client connected to the Elasticsearch node. With Spring Data, the High Level REST Client is the default client, although Elasticsearch documentation states that it's been deprecated in favor of the Java API Client since version 7.15. The Java API Client is not listed as a supported client yet. In the meantime, the Java Transport Client is still supported in Spring Data, but the general recommendation is to use the High Level Client.
```
clarity